daemon/containerd/image_delete.go

```diff
@@ -35,6 +41,24 @@ import "github.com/docker/docker/api/types"
 // If prune is true, ancestor images will each attempt to be deleted quietly,
 // meaning any delete conflicts will cause the image to not be deleted and the
 // conflict will not be reported.
-func (i *ImageService) ImageDelete(imageRef string, force, prune bool) ([]types.ImageDeleteResponseItem, error) {
-	panic("not implemented")
+func (i *ImageService) ImageDelete(ctx context.Context, imageRef string, force, prune bool) ([]types.ImageDeleteResponseItem, error) {
+	// TODO(thaJeztah): implement ImageDelete "force" options; see https://github.com/moby/moby/issues/43850
+	// TODO(thaJeztah): implement ImageDelete "prune" options; see https://github.com/moby/moby/issues/43849
+
+	records := []types.ImageDeleteResponseItem{}
+
+	parsedRef, err := reference.ParseNormalizedNamed(imageRef)
+	if err != nil {
+		return nil, err
+	}
+	ref := reference.TagNameOnly(parsedRef)
+
+	if err := i.client.ImageService().Delete(ctx, ref.String(), images.SynchronousDelete()); err != nil {
+		return []types.ImageDeleteResponseItem{}, err
```

Comment:
How come an empty slice is returned here but the error-return for `ParseNotmalizedNamed` (line 52) returns `nil`?

Reply:
Good catch; at a glance, I think can be just `nil` as well, unless there's some expectation somewhere (`nil` vs `nil type`)? Overall the returned value shouldn't be used if there's an error.

Let me have a look at that.